scipy/optimize/_differentiable_functions.py

```diff
@@ -523,34 +536,38 @@ def update_jac():
                 self.J = jac_wrapped(self.x)
 
         elif jac in FD_METHODS:
-            self.J = approx_derivative(fun_wrapped, self.x, f0=self.f,
-                                       **finite_diff_options)
+            self.J, dct = approx_derivative(fun_wrapped, self.x, f0=self.f,
+                                            **finite_diff_options)
             self.J_updated = True
+            self.nfev += dct['nfev']
 
             if (sparse_jacobian or
                     sparse_jacobian is None and sps.issparse(self.J)):
                 def update_jac():
                     self._update_fun()
-                    self.J = sps.csr_array(
+                    self.J, dct = sps.csr_array(
                         approx_derivative(fun_wrapped, self.x, f0=self.f,
                                           **finite_diff_options))
+                    self.nfev += dct['nfev']
                 self.J = sps.csr_array(self.J)
                 self.sparse_jacobian = True
 
             elif sps.issparse(self.J):
                 def update_jac():
                     self._update_fun()
-                    self.J = approx_derivative(fun_wrapped, self.x, f0=self.f,
-                                               **finite_diff_options).toarray()
+                    self.J, dct = approx_derivative(fun_wrapped, self.x, f0=self.f,
+                                                    **finite_diff_options).toarray()
+                    self.nfev += dct['nfev']
```

Comment:
Could you elaborate a bit on this? I might not understand fully but was `nfev` actually counted correctly before this PR without parallelization?

Reply:
It was counted correctly. The main change is that previously there were wrappers of the user's objective function that tracked the number of calls by storing it in an attribute. These wrappers (converted to classes) can now be evaluated in parallel in different processes in `approx_derivative`. This parallel evaluation means that the attribute can't be updated properly. Therefore `approx_derivative` calculates what the number of function evaluations would've been and returns it in the dictionary.

Reply:
The code calling `approx_derivative` then has to keep a total tally of func evaluations. The tests for total number of function evaluations are quite strict for `_differentiable_functions`.